libs/secure-sm/src/test/java/org/elasticsearch/secure_sm/SecureSMTests.java

```diff
@@ -20,8 +24,13 @@
 import java.util.stream.Collectors;
 
 /** Simple tests for SecureSM */
-public class SecureSMTests extends TestCase {
```

Comment:
why is TestCase removed?

Reply:
I needed to upgrade from JUnit3 (TestCase) to JUnit4 (only annotations) to be able to use `@BeforeClass`
A static initializer like it was before will not do - assumeFalse will throw, but we will have an InvocationTargetException or similar, and the test will really fail instead of being treated as ignored

Reply:
We use `@BeforeClass` in other places. I suggest making this extend Assert (that's what LuceneTestCase does), and add `@TestMethodProviders({JUnit3MethodProvider.class})` to the class